website/docs/flow/context/index.md

```diff
@@ -0,0 +1,129 @@
+
+`pending_user` is used by multiple stages. In the context of most flow executions, it holds the user that is executing the flow. This value is not set automatically, it is set via the [Identification stage](../stages/identification/).
+
+Stages that require a user, such as the [Password stage](../stages/password/), the [Authenticator validation stage](../stages/authenticator_validate/) and others will use this value if it is set, and fallback to the request's users when possible.
+
+### `prompt_data` (Dictionary)
+
+`prompt_data` is primarily used by the [Prompt stage](../stages/prompt/). The value of any field within a prompt stage is written to the `prompt_data` dictionary. For example, given a field with the _Field key_ `email` that was submitted with the value `foo@bar.baz` will result in the following context:
+
+```json
+{
+    "prompt_data": {
+        "email": "foo@bar.baz"
+    }
+}
+```
+
+This data can be modified with policies. The data is also used by stages like [User write](../stages/user_write.md), which takes data in `prompt_data` and writes it to `pending_user`.
+
+### `redirect` (string)
+
+Stores the final redirect that the user's browser will be sent to after the flow is finished executing successfully. This is set when an un-authenticated user attempts to access a secured application, and when a user authenticates/enrolls with an external source.
+
+### `application` (Application object)
+
+When an unauthenticated user attempts to access a secured resource, they are redirected to an authentication flow. The application they attempted to access will be stored in this key.
+
+### `source` (Source object)
+
+When a user authenticates/enrolls via an external source, this will be set to the source they are using.
+
+## Scenario specific
+
+### `is_sso` (boolean)
+
+Set to `True` when the flow is executed from an "SSO" context. For example, this is set when a flow is used during the authentication or enrollment via an external source, and if a flow is executed to authorize access to an application.
+
+## Stage specific
+
+### Consent stage
+
+#### `consent_header` (string)
+
+The title of the consent prompt shown. Set automatically when the consent stage is used with a OAuth2, Proxy or SAML provider.
+
+#### `consent_permissions` (List of PermissionDict)
+
+An optional list of all permissions that will be given to the application by granting consent. Not supported with SAML. When used with an OAuth2 or Proxy provider, this will be set based on the configured scopes.
+
+### Autosubmit stage
+
+The autosubmit stage is an internal stage type that is not configurable via the API/Web interface. It is used in certain situations, where a POST request has to be sent from the browser, such as with SAML POST bindings. This works by using an HTML form that is submitted automatically.
+
+#### `title` (string)
+
+Optional title of the form shown to the user. Automatically set when this stage is used by the backend.
```

Comment:
Automatically set to what value? Does user ever see it?

Reply:
Usually in a format of `Redirecting to $app...`. The user sees it very briefly as the redirect happens

This stage is a bit different as an admin can't configure it via the API/web interface, it's an "in-memory" stage that gets added by authentik itself when it's required